core/jvm/src/main/scala/zio/metrics/jvm/Thread.scala

```diff
@@ -89,13 +89,15 @@ object Thread {
           )
 
         schedule <- ZIO.service[JvmMetricsSchedule]
-        _        <- threadsCurrent.launch(schedule.value)
-        _        <- threadsDaemon.launch(schedule.value)
-        _        <- threadsPeak.launch(schedule.value)
-        _        <- threadsStartedTotal.launch(schedule.value)
-        _        <- threadsDeadlocked.launch(schedule.value)
-        _        <- threadsDeadlockedMonitor.launch(schedule.value)
-        _        <- ZIO.acquireRelease(refreshThreadStateCounts(threadMXBean).repeat(schedule.value).forkDaemon)(_.interrupt)
+        _        <- threadsCurrent.launch(schedule.updateMetrics)
+        _        <- threadsDaemon.launch(schedule.updateMetrics)
+        _        <- threadsPeak.launch(schedule.updateMetrics)
+        _        <- threadsStartedTotal.launch(schedule.updateMetrics)
+        _        <- threadsDeadlocked.launch(schedule.updateMetrics)
+        _        <- threadsDeadlockedMonitor.launch(schedule.updateMetrics)
+        _ <- ZIO.acquireRelease(refreshThreadStateCounts(threadMXBean).repeat(schedule.updateMetrics).forkDaemon)(
```

Comment:
Should this be: 


```scala
ZIO.acquireRelease(ZIO.interruptibly(refreshThreadStateCounts(threadMXBean).repeat(schedule.updateMetrics).)forkDaemon)(
```

Otherwise it will hang shutdown because it can't be interrupted. Please make this change for all the other `ZIO.acquireRelease` ones. 

Also, maybe we are missing something to accomplish this?

e.g. `.scheduleBackground` which could return `ZIO[Scope, ...]` and handle this messy part.